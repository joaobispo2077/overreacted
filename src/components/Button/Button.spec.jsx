import { Button } from '.';
import userEvent from '@testing-library/user-event';
const { render, screen, fireEvent } = require("@testing-library/react");

describe('<Button />', () => {
  it('should render the button with text "Load more"', () => {
    render(<Button text="Load more" />);

    const button = screen.getByRole('button', { name: /load more/i });
    expect(button).toBeInTheDocument();
  });

  it('should call on click function', () => {
    const mockOnClick = jest.fn();

    render(<Button text="Load more" onClick={mockOnClick} />);

    const button = screen.getByRole('button', { name: /load more/i });

    userEvent.click(button) // naturally event
    fireEvent.click(button); // just trigger event

    expect(mockOnClick).toHaveBeenCalledTimes(2);
  });

  it('should be disabled when disabled is true', () => {
    render(<Button text="Load more" disabled={true} />);

    const button = screen.getByRole('button', { name: /load more/i });

    expect(button).toBeDisabled();
  });

  it('should be disabled when disabled is false', () => {
    render(<Button text="Load more" disabled={false} />);

    const button = screen.getByRole('button', { name: /load more/i });

    expect(button).toBeEnabled();
  });

  it('shoud match snapshot', () => {
    const mockOnClick = jest.fn();
    const { container } = render(<Button text="Load more" onClick={mockOnClick} />);

    expect(container.firstChild).toMatchSnapshot();
  });
});