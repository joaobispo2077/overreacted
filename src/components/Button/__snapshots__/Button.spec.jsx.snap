// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Button /> shoud match snapshot 1`] = `
<button
  class="button"
>
  Load more
</button>
`;
